src/isl/isl_type.rs

```diff
@@ -119,3 +119,18 @@ impl IslTypeImpl {
         Ok(IslTypeImpl::new(type_name, constraints))
     }
 }
+
+// OwnedStruct doesn't preserve field order hence the PartialEq won't work properly for unordered constraints
+// Related issue: https://github.com/amzn/ion-rust/issues/200
+impl PartialEq for IslTypeImpl {
```

Comment:
non-blocking question—can you explain why we're using `PartialEq` instead of `Eq` for all of the types in this library? As far as I can tell, the only difference is reflexivity, and I can't think of any case for ISL where `a != a`. (I know that `nan` exists and `nan != nan`, but in the context of ISL, something like `valid_values: [ X ]` has reflexive equality even if `X` is `nan`.)